components/nimbus/src/lib.rs

```diff
@@ -210,25 +238,140 @@ impl NimbusClient {
+        log::info!("[Nimbus] Update Date: {}", update_date);
+        let now = Utc::now();
+        let duration_since_install = now - installation_date;
+        log::info!(
+            "[Nimbus] Days since install: {}",
+            duration_since_install.num_days()
+        );
+        let duration_since_update = now - update_date;
+        log::info!(
+            "[Nimbus] Days since update: {}",
+            duration_since_update.num_days()
+        );
+        let mut state = self.mutable_state.lock().unwrap();
+        if state.targeting_attributes.days_since_install.is_none() {
+            state.targeting_attributes.days_since_install =
+                Some(duration_since_install.num_days() as i32);
+        }
+        if state.targeting_attributes.days_since_update.is_none() {
+            state.targeting_attributes.days_since_update =
+                Some(duration_since_update.num_days() as i32);
+        }
+
         let pending_updates = read_and_remove_pending_experiments(db, &mut writer)?;
-        Ok(match pending_updates {
+        let res = match pending_updates {
             Some(new_experiments) => {
                 let nimbus_id = self.read_or_create_nimbus_id(db, &mut writer)?;
-                let state = self.mutable_state.lock().unwrap();
                 let evolver = EnrollmentsEvolver::new(
                     &nimbus_id,
                     &state.available_randomization_units,
-                    &self.app_context,
+                    &state.targeting_attributes,
                 );
-                let events = evolver.evolve_enrollments_in_db(db, &mut writer, &new_experiments)?;
-                self.database_cache.commit_and_update(db, writer)?;
-                events
+                evolver.evolve_enrollments_in_db(db, &mut writer, &new_experiments)?
             }
-            // We don't need to writer.commit() here because we haven't done anything.
             None => vec![],
-        })
+        };
+        self.database_cache.commit_and_update(db, writer)?;
+        Ok(res)
+    }
+
+    fn get_installation_date(&self, db: &Database, writer: &mut Writer) -> Result<DateTime<Utc>> {
+        // we first check our context
+        if let Some(context_installation_date) = self.app_context.installation_date {
+            let res = DateTime::<Utc>::from_utc(
+                NaiveDateTime::from_timestamp(context_installation_date / 1_000, 0),
+                Utc,
+            );
+            log::info!("[Nimbus] Retrieved date from Context: {}", res);
+            return Ok(res);
+        }
+        let store = db.get_store(StoreId::Meta);
+        let persisted_installation_date: Option<DateTime<Utc>> =
+            store.get(writer, DB_KEY_INSTALLATION_DATE)?;
+        let res = if let Some(installation_date) = persisted_installation_date {
+            installation_date
+        } else {
+            let installation_date = match self.get_creation_date_from_path(&self.db_path) {
```

Comment:
Hmmmm good catch, I wrongly assumed that the db_path is something that was created even before nimbus came by

This is mainly a problem with old installations of iOS apps, or new iOS apps... There are two paths forward for me, I'll timebox a little and pick one
- Have the apps pass the home directory
- Have iOS pass an installation time like android, might need to figure out a good API for that - if I find one, we can completely eliminate the `get_creation_date_from_path`!

Reply:
Used `NSHomeDirectory`! (also added a home directory for android using `context.applicationInfo?.dataDir`, but for the purposes of this PR, it won't be used since android can pass in the installation date directly)